core/controllers/base.py

```diff
@@ -57,14 +58,14 @@ def _clear_login_cookies(response_headers):
     # App Engine sets the ACSID cookie for http:// and the SACSID cookie
     # for https:// . We just unset both below.
     cookie = http.cookies.SimpleCookie()
-    for cookie_name in ['ACSID', 'SACSID']:
+    for cookie_name in [b'ACSID', b'SACSID']:
```

Comment:
One question. For most of these, do they actually _have_ to be b'...'? If they can be unicode strings, and the prefix is not needed, we should probably allow them to be unicode strings?

Reply:
Yes I have marked only those strings which require to be in b''

Reply:
I checked them manually. The tests don't run if I remove any of the b'' literal in base.py

Reply:
This is the traceback that I get on running the tests:
```
response = self.get_html_response('/about', expected_status_int=302)
  File "/home/jude/foss/oppia/core/tests/test_utils.py", line 582, in get_html_response
    expected_status_int=expected_status_int)
  File "/home/jude/foss/oppia/core/tests/test_utils.py", line 548, in _get_response
    status=expected_status_int)
  File "/home/jude/foss/oppia/../oppia_tools/webtest-2.0.33/webtest/app.py", line 335, in get
    expect_errors=expect_errors)
  File "/home/jude/foss/oppia/../oppia_tools/webtest-2.0.33/webtest/app.py", line 654, in do_request
    self._check_status(status, res)
  File "/home/jude/foss/oppia/../oppia_tools/webtest-2.0.33/webtest/app.py", line 689, in _check_status
    "Bad response: %s (not %s)\n%s", res_status, status, res)
AppError: Bad response: 500 Internal Server Error (not 302)
500 Internal Server Error

The server has either erred or is incapable of performing the requested operation.
```
That's why these has to be in bytes